src/lib/ui/containers/SignupForm/SignupForm.svelte

```diff
@@ -0,0 +1,158 @@
+  import { getTexts } from "$lib/logic/utils/objects.js";
+  import { fieldsSignup } from "$lib/logic/schemas/index.js";
+
+  import type { Readable } from "svelte/store";
+  import type { FieldInputForm } from "$lib/logic/typing/globals/interfaces.js";
+  import type { SignupValues } from "$lib/logic/typing/schemas/auth.js";
+  import type { ConfigForm } from "$lib/logic/typing/store/config.js";
+  import type { Props, SignupFields } from "./SignupForm.proptypes.js";
+
+  import * as stylesinternal from "./SignupForm.styles.js";
+
+  export let submit: Props["submit"];
+  export let context: Props["context"] = "form";
+  export let showErrors: Props["showErrors"] = true;
+  export let options: Props["options"] = [];
+  export let code: Props["code"] = "bycountry";
+  export let confirm: Props["confirm"] = true;
+  export let styles: Props["styles"] = undefined;
+  export let texts: Props["texts"];
+
+  const globalStyles = getContext<Readable<ConfigForm["form"]>>("formStyles");
+  $: formStyles = styles?.form ?? $globalStyles ?? stylesinternal ?? {};
+  const { confirmPassword, ...fieldsRemaining } = fieldsSignup;
+
+  $: store = faivform({
+    fields: { ...fieldsRemaining, ...(confirm ? { confirmPassword } : {}) },
+    styles: {
+      field: styles?.field ?? {},
+      select: styles?.select ?? {},
+      icons: styles?.icons ?? null,
+    },
+  });
+  $: ({ submit: onSubmit, setField, setError, loading } = $store);
+  const dispatch = createEventDispatcher<{
+    error: unknown;
+    finish: undefined;
+    choose: {
+      value: string;
+      setField: typeof setField;
+      setError: typeof setError;
+    };
+  }>();
+
+  function onChoose({ detail: value }: CustomEvent<string>) {
+    dispatch("choose", { value, setField, setError });
+  }
+
+  $: action = onSubmit<SignupValues>(
+    async (values) => {
+      if (confirm && values.password.trim() !== values.confirmPassword.trim()) {
+        const error = "password-not-match";
+        setError("password", error);
+        setError("confirmPassword", error);
+        throw new FormError({
+          title: "Error to signup",
+          message: "Password and confirm password must be the same",
+          reason: error,
+        });
+      }
+      await submit(values);
+    },
+    {
+      error(error) {
+        dispatch("error", error);
+      },
+      finish: () => {
+        dispatch("finish");
+      },
+      context,
+    }
+  );
+
+  $: fields = (
+    [
+      {
+        name: "firstName",
+        type: "text",
+      },
+      {
+        name: "lastName",
+        type: "text",
+      },
+      {
+        name: code === "bycountry" ? "country" : "phoneCode",
+        type: "select",
+      },
+      {
+        name: "email",
+        type: "email",
+      },
+      {
+        name: "phone",
+        type: "tel",
+      },
+      {
+        name: "password",
+        type: "password",
+      },
+      ...(confirm
+        ? ([
+            {
```

Comment:
🚫 **[eslint]** <[@typescript-eslint/indent](https://typescript-eslint.io/rules/indent)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expected indentation of 10 spaces but found 12.
```suggestion
          {
```